libs/platform/ebpf_bitmap.h

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+#pragma once
+#include <ebpf_platform.h>
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+    typedef struct _ebpf_bitmap ebpf_bitmap_t;
+    typedef uintptr_t ebpf_bitmap_cursor_t[3];
+
+    /**
+     * @brief Compute the size in bytes required to hold a ebpf_bitmap_t.
+     *
+     * @param[in] bit_count Count of bits the bitmap will hold.
+     * @return Size in bytes required.
+     */
+    size_t
+    ebpf_bitmap_size(size_t bit_count);
+
+    /**
+     * @brief Initialize an already allocated bitmap.
+     *
+     * @param[out] bitmap Pointer to the bitmap.
+     * @param[in] bit_count Count of bits to be stored.
+     */
+    void
+    ebpf_bitmap_initialize(_Out_ ebpf_bitmap_t* bitmap, size_t bit_count);
+
+    /**
+     * @brief Set bit at index to true.
+     *
+     * @param[in] bitmap Pointer to the bitmap.
+     * @param[in] index Index to modify.
+     * @param[in] interlocked Perform the operation using interlocked.
+     * @retval true of the bit was set, false otherwise.
+     */
+    bool
+    ebpf_bitmap_set_bit(_Inout_ ebpf_bitmap_t* bitmap, size_t index, bool interlocked);
+
+    /**
+     * @brief Set bit at index to false.
+     *
+     * @param[in] bitmap Pointer to the bitmap.
+     * @param[in] index Index to modify.
+     * @param[in] interlocked Perform the operation using interlocked.
+     * @retval true of the bit was set, false otherwise.
+     */
+    bool
+    ebpf_bitmap_reset_bit(_Inout_ ebpf_bitmap_t* bitmap, size_t index, bool interlocked);
+
+    /**
+     * @brief Get the value of the bit at index.
+     *
+     * @param[in] bitmap Pointer to the bitmap.
+     * @param[in] index Index to modify.
+     * @retval true of the bit was set, false otherwise.
```

Comment:
```suggestion
     * @return true if the bit was set, false otherwise.
```
see earlier comment about about return vs retval